core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -1,22 +1,214 @@
 package com.databricks.labs.remorph.parsers.tsql
 
-import org.antlr.v4.runtime.Token
-import org.antlr.v4.runtime.tree.TerminalNode
-
-import com.databricks.labs.remorph.parsers.intermediate.{Column, Literal}
+import com.databricks.labs.remorph.parsers.intermediate.{Column, Expression, Literal}
 import com.databricks.labs.remorph.parsers.tsql.TSqlParser._
 import com.databricks.labs.remorph.parsers.{IncompleteParser, ParserCommon, intermediate => ir}
+import org.antlr.v4.runtime.Token
+import org.antlr.v4.runtime.tree.TerminalNode
 
+import java.util.Locale
 import scala.collection.JavaConverters._
 
 class TSqlExpressionBuilder
     extends TSqlParserBaseVisitor[ir.Expression]
     with ParserCommon
     with IncompleteParser[ir.Expression] {
 
+  case class FunDef(argMin: Int, argMax: Int = Int.MaxValue, convertible: Boolean = true)
```

Comment:
`Arity` or `FunctionArity` may be more informative names than `FunDef`. Looking at the usage bellow, it may also benefit from being defined as an ADT:

```scala
sealed trait FunctionArity
case class FixedArity(nbParameters: Int) extends FunctionArity
case class VariableArity(argMin: Int, argMax: Int) extends FunctionArity
```

The `convertible` parameter seems to always be set to its default value, so I ignored it for now.

Reply:
> `Arity` or `FunctionArity` may be more informative names than `FunDef`. Looking at the usage bellow, it may also benefit from being defined as an ADT:
> 
> ```
> sealed trait FunctionArity
> case class FixedArity(nbParameters: Int) extends FunctionArity
> case class VariableArity(argMin: Int, argMax: Int) extends FunctionArity
> ```
> 
> The `convertible` parameter seems to always be set to its default value, so I ignored it for now.

I don't mind changing the name but I think it is likely that this table may expand to have information other than arity. It does have a parameter that indicates whether we can convert it for instance. I guess we can change it again later  so I will change to use this trait pattern.